### PR TITLE
Add rule to check Do Not Translate terms

### PR DESCRIPTION
**Pull request by wadimw**
Create a rule that checks if the term exists in the source and if so, the same term appears somewhere in the target.

### CODE REVIEW

File: src/rules/ResourceDNTTerms.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ * limitations under the License.
+ */
+
+import { Rule, Result } from "i18nlint-common";
+import fs from "node:fs";
+
+/** Rule to ensure that Do Not Translate terms have not been translated;
+ * i.e., if a DNT term appears in source, it has to appear in target as well */
+class ResourceDNTTerms extends Rule {
+    /** @readonly */ name = "resource-dnt-terms";
```
Comment (ehoogerbeets) [inline src/rules/ResourceDNTTerms.js:26]: To make it readonly, use `#name = "resource-dnt-terms";`

File: src/rules/ResourceDNTTerms.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    /** @readonly */ ruleType = "resource";
+
+    /**
+     * @readonly
+     * @type {string[]}
+     */
+    _dntTerms = [];
+
+    get dntTerms() {
+        return [...this._dntTerms];
```
Comment (ehoogerbeets) [inline src/rules/ResourceDNTTerms.js:38]: Why do we need a getter for this array?

File: test/testRules.js [hunk 3]
```diff
@@ -1674,5 +1675,193 @@ export const testRules = {
...
+        });
+        test.ok(rule);
+
+        test.deepEqual(rule.dntTerms, [
+            "Some DNT term",
+            "Another DNT term"
+        ]);
+
+        test.done();
+    },
```
Comment (ehoogerbeets) [inline test/testRules.js:1865]: Please add some tests to check for DNT terms in ResourceArray and ResourcePlural resources.

File: src/rules/ResourceDNTTerms.js (hunk 1 above)
Comment (wadimw) [reply src/rules/ResourceDNTTerms.js:26]: Actually, `#` makes it private https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields.

To make it truly readonly, one could probably use `Object.defineProperty` with `writeable: false`, but I feel like it does not matter _that much_ here to warrant the boilerplate - I just put the JSDoc annotation for the IntelliSense/completion purposes mostly

File: src/rules/ResourceDNTTerms.js (hunk 2 above)
Comment (wadimw) [reply src/rules/ResourceDNTTerms.js:38]: That one I exposed on a whim to be able to unittest if terms were loaded correctly. I guess it's not really needed, so I'll just change the static parsing methods to be public and test them directly

Comment (wadimw) [reply src/rules/ResourceDNTTerms.js]: Changed in 1ea955f

File: src/rules/ResourceDNTTerms.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+     * @param {string} options.file
+     * @param {string} options.locale
+     *
+     */
+    match({ resource, file, locale }) {
+        const source = resource.getSource();
+        const target = resource.getTarget();
+
+        if ("string" !== typeof source || "string" !== typeof target) {
+            return /* don't check when either source or target string is not defined */;
```
[the remark this answers is not in the document]
Comment (wadimw) [reply src/rules/ResourceDNTTerms.js]: Changed in f3570d7

File: test/testfiles/dnt-test.txt [hunk 5]
```diff
@@ -1,2 +1,6 @@
 Some DNT term
 Another DNT term
+    A DNT term that should be trimmed
+Yet another DNT temr that should be trimmed
```
Comment (ehoogerbeets) [inline test/testfiles/dnt-test.txt:4]: spelling `temr`

File: src/rules/ResourceDNTTerms.js [hunk 6]
```diff
@@ -66,10 +64,10 @@ class ResourceDNTTerms extends Rule {
             // if the terms are not provided explicitly, parse them from a file
             switch (params.termsFileType) {
```
[the remark this answers is not in the document]
Comment (wadimw) [reply src/rules/ResourceDNTTerms.js]: Implemented in e822628

File: test/testfiles/dnt-test.txt (hunk 5 above)
Comment (wadimw) [reply test/testfiles/dnt-test.txt]: Fixed in 2a21c5a

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**